src/python/CMSSpark/wmarchive_crab_file_access.py

```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+File        : wmarchive_crab_file_access.py
+Author      : Author: Ceyhun Uzunoglu <ceyhunuzngl AT gmail [DOT] com>
+Description : Generates last/first access of datasets using WMArchive data belonging CRAB jobs
+
+How:
+  - Only filter is that meta_data.jobtype should start with CRAB for the aim of just to get user jobs' accesses to files
+  - data.meta_data.ts is used as file access time
+  - files' datasets are extracted from join with DBS tables and additional DBS dataset info is placed in each row
+  - Main calculation is to get LastAccess and FirstAccess of datasets which are only accessed by user jobs
+"""
+
+# system modules
+import json
+import logging
+import os
+import sys
+import time
+from datetime import date, datetime, timezone
+
+import click
+from pyspark.sql.functions import (
+    col as _col,
+    countDistinct,
+    first,
+    max as _max,
+    min as _min,
+    sum as _sum,
+)
+from pyspark.sql.types import ArrayType, StructType, StructField, StringType, LongType
+
+# CMSSpark modules
+from CMSSpark.spark_utils import get_spark_session, get_candidate_files
+
+# CMSMonitoring modules
+try:
+    from CMSMonitoring.StompAMQ7 import StompAMQ7
+except ImportError:
+    print("ERROR: Could not import StompAMQ")
+    sys.exit(1)
+
+# global variables
+_DEFAULT_HDFS_FOLDER = "/project/monitoring/archive/wmarchive/raw/metric"
+_VALID_DATE_FORMATS = ["%Y/%m/%d", "%Y-%m-%d", "%Y%m%d"]
+
+# global variables
+TODAY = datetime.today().strftime('%Y-%m-%d')
+# DBS
+HDFS_DBS_DATASETS = f'/tmp/cmsmonit/rucio_daily_stats-{TODAY}/DATASETS/part*.avro'
+HDFS_DBS_FILES = f'/tmp/cmsmonit/rucio_daily_stats-{TODAY}/FILES/part*.avro'
+HDFS_DBS_DATA_TIERS = f'/tmp/cmsmonit/rucio_daily_stats-{TODAY}/DATA_TIERS/part*.avro'
+HDFS_DBS_PHYSICS_GROUPS = f'/tmp/cmsmonit/rucio_daily_stats-{TODAY}/PHYSICS_GROUPS/part*.avro'
+HDFS_DBS_ACQUISITION_ERAS = f'/tmp/cmsmonit/rucio_daily_stats-{TODAY}/ACQUISITION_ERAS/part*.avro'
+HDFS_DBS_DATASET_ACCESS_TYPES = f'/tmp/cmsmonit/rucio_daily_stats-{TODAY}/DATASET_ACCESS_TYPES/part*.avro'
+
```

Comment:
I suggest to add dump of all global variable to the stdout to help debugging process